developing/vcs/README.md

```diff
@@ -2,7 +2,94 @@
 
 ## Overview
 
-Source control has less to do with the vcs system you use and more to do with the stages code goes through on the journey from inside a developers head, tested and merged into the main line, deployed to production and thence in time to being identified as that accursed legacy system that is causing all the problems.
+Source control has less to do with the VCS system you use, and more to do with the stages code goes through on the journey from inside a developer's head, tested and merged into the main line, deployed to production, and thence in time to being identified as that accursed legacy system that is causing all the problems.
+
+## Repo maintenance
+
+* Avoid including files in source control that are specific to your development machine or process.
+* Ignore files that are specific to your machine or process in your `~/.gitignore_global` as opposed to your repo's `.gitignore`.
+* Delete local and remote feature branches after merging.
+* Perform work in a feature branch.
+* Rebase frequently to incorporate upstream changes.
+* Use a pull request for code reviews.
+
+### GitHub repo settings
+
+* [Protect the master branch](https://help.github.com/en/articles/configuring-protected-branches) by turning on the follow settings at a minimum:
+
+  * Require pull request reviews before merging
+  * Require status checks to pass before merging
+
+## Git workflow
+
+Clone the repo locally with SSH:
+
+```
+git clone git@github.com:trussworks/Engineering-Playbook.git
+cd Engineering-Playbook
+```
+
+Create a local feature branch based off master. Start the branch name with your
+initials. For example: `mb-git-worfklow`. If the branch is related to a Pivotal
+story,
+
+```
+git checkout master
+git pull
+git checkout -b <your-initials-branch-name>
+```
+
+Rebase frequently to incorporate upstream changes.
+
+```
+git fetch origin
+git rebase origin/master
+```
+
+Resolve conflicts. When the feature/fix is complete and tests pass, stage the changes.
+
+```
+git add --all
+```
+
+When you've staged the changes, commit them.
+
+```
+git status
+git commit --verbose
+```
+
+Write a [good commit message](https://chris.beams.io/posts/git-commit/). Example
+format:
+
+```
+Summary under 50 characters in imperative mood
+
+* Why this commit is necessary (under 72 characters).
+* More information about commit (under 72 characters).
+* More information about commit (under 72 characters).
+
+Resolves #123 if GitHub issue or link to Pivotal story
+```
+
+To help enforce this commit message format, it's helpful to add our pull request
+template in your repo.
+
+If you've created more than one commit, use `[git rebase interactively](https://help.github.com/en/articles/about-git-rebase)` to squash them into cohesive commits with good messages:
```

Comment:
I don't think there has been a poll recently but this seems like a lightning rod to me.

Reply:
Interesting. I did not realize this was a controversial subject. I plan on expanding this to specify that this applies to noisy commits that don't add much value, such as fixing typos, lint errors, or commits made while testing a third party integration. For example, when trying to configure Circle CI, one might try different things before getting it right. I don't think it makes sense to keep all the commits that didn't work. 

Do folks have strong opinions in favor of keeping noisy commits as well?

I think that squashing into meaningful and concise commits clarifies the intent and purpose of individual commits when looking through history.

Reply:
IME, I've usually opted for keeping the noisy stuff. Partially that was laziness, partially has to do with wanting to remind myself how the sausage was made, partially has to do with the cleanup workflow on my machine... (Not having to do a _force_ delete of the branch that I pushed up, e.g. `git branch -d examplebranch` vs `git branch -D examplebranch`...)

Yet I am not feeling super married to it, so I'm open to trying it the squash way too.

Reply:
> I plan on expanding this to specify that this applies to noisy commits that don't add much value, such as fixing typos, lint errors, or commits made while testing a third party integration. For example, when trying to configure Circle CI, one might try different things before getting it right. I don't think it makes sense to keep all the commits that didn't work.

I don't see any issue with rebasing commits that haven't been pushed yet. If they have been pushed, then the workflow may invove routine use of `push --force`, which is sketchy imo.

Regardless, I should have been more explicit. The lightning rod, in my view, is about rebase vs merge.

Reply:
This is great. It has given me more ideas on what to write about. There is a way to squash commits and still be able to use `-d`. As for force pushing, that would only happen once, when the PR is ready to be merged.

Reply:
> As for force pushing, that would only happen once, when the PR is ready to be merged.

I understand. I think we should have a consensus that this is a best practice before we publish it as best practice. It has been awhile since there was broad discussion about it but I believe I'm not alone in having reservations about it.